homeassistant/components/sonos/media_player.py

```diff
@@ -1245,4 +1263,7 @@ def device_state_attributes(self):
         if self._speech_enhance is not None:
             attributes[ATTR_SPEECH_ENHANCE] = self._speech_enhance
 
+        if self._queue_position is not None and self._is_playing_local_queue:
+            attributes[ATTR_QUEUE_POSITION] = self._queue_position
```

Comment:
There is now no `queue_position` for grouped players, except for the master. Is this intended?

You can make a property that forwards to the coordinator as needed like this:
```python
    @property
    @soco_coordinator
    def queue_position(self):
        """Current position in the playing queue."""
        return self._queue_position
```
and then use the non-underscored attribute when reading the value.

Otherwise, I think it looks good now 👍

Reply:
This was intended. Because only the coordinator is only playing local queue.

Reply:
Okay. I find it a bit surprising as slaves do have e.g. `media_title` from the coordinator.

Reply:
@amelchio 
Are you OK with this changes?

Reply:
Is there a downside to reporting the same `queue_position` on all devices in a group and not just the coordinator? From a user's standpoint it seems confusing when most other attributes/commands work on any device in the group.

Reply:
From a usage standpoint, aren't the grouped players' queues effectively "hidden" while they're not the coordinator? Shouldn't queue operations operate on the group coordinator's currently playing queue and not a queue that _will_ be in use when it's later ungrouped?

Reply:
I didn't try this, but, when you have one automation that covers all your sonos, to remove the last played song from the queue, and you have a group of 3 players, I should think there were 3 songs removed from the queue since every player forwards that service to the coordinator.
Or there should be a condition in the automation to only trigger this automation on a coordinator, lets say ```ATTR_IS_COORDINATOR = True```. 
Or leave it this way only give the coordinator  the ```queue_position```, then you can use 
```
{% if states.media_player.kitchen.attributes.queue_position is defined and states.media_player.~player~.attributes.queue_position >= 1 %}
```
And lets say
```

    @soco_error()
    def play_queue(self, queue_position=0):
         """Start playing the queue."""
         self.soco.play_from_queue(queue_position)
```
Is also not forwarded to the coordinator

Reply:
I agree with @jjlawren that for consistency it is better to have the new attribute duplicated on slave speakers since we do that with all the other attributes.

It is already possible to identify coordinators because their own `entity_id` will be the first one listed in the `sonos_group` attribute. That will probably not be more complicated than checking for presence of `queue_position`?

You are right about `play_queue` but I think that is a bug since it cannot actually be called on the slave:

```
Error on play_queue with The method or property "play_from_queue" can only be called/used on the coordinator in a group
```

So we should probably add `@soco_coordinator` to both `remove_from_queue` and `play_queue`.

Reply:
Now the ```queue_position``` is -1 when not playing local queue, because otherwise the template in the condition always trows an error when restoring from a snapshot

Reply:
You can probably avoid that error by using
```jinja
{% if state_attr('media_player.kitchen', 'queue_position') %}
```
as `state_attr()` will return `None` if either the entity or the attribute is missing. Using -1 breaks that standard behavior so I think we should avoid it.

Reply:
Can I show the attribute with value None? Instead of removing it when not playing local?

Reply:
```python
     @property
     @soco_coordinator
     def queue_position(self):
         """If playing local queue return the position in the queue else -1."""
         if self._is_playing_local_queue:
             return self._queue_position

         return None
```
```python
        attributes[ATTR_QUEUE_POSITION] = self.queue_position
```

Reply:
I am quite sure there is a rule against `None` attributes, so unfortunately no.

Reply:
```
condition:
  condition: and
  conditions:
    - condition: template
      value_template: >
        {% set group = state_attr( trigger.entity_id , 'sonos_group') %}
        {% if group[0] == trigger.entity_id %}
          True
        {% else %}
          False
        {% endif %}
    - condition: template
      value_template: >
        {% if state_attr( trigger.entity_id, 'queue_position') %}
          {% if state_attr( trigger.entity_id, 'queue_position') >= 1 %}
            {{ trigger.from_state.attributes.queue_position < trigger.to_state.attributes.queue_position }}
          {% else %}
            False
          {% endif %}
        {% else %}
          False
        {% endif %}
```
When restoring from snapshot
```
Error during template condition: UndefinedError: 'mappingproxy object' has no attribute 'queue_position'
```

Reply:
Can you think of a way to have the integration handle this instead of requiring special case handling in every script/automation?

Reply:
Yeah okay, that is a bit more complex with the from/to states. I think something like this should work (tested somewhat):
```yaml
    condition:
      # Coordinator
      - condition: template
        value_template: >
          {{ state_attr( trigger.entity_id , 'sonos_group')[0] ==  trigger.entity_id }}
      # Going from queue to queue
      - condition: template
        value_template: >
          {{ 'queue_position' in trigger.from_state.attributes and 'queue_position' in trigger.to_state.attributes }}
      # Moving forward
      - condition: template
        value_template: >
          {{ trigger.from_state.attributes.queue_position < trigger.to_state.attributes.queue_position }}
```

... or you can use `trigger.from_state.attributes.get('queue_position', -1)` to emulate the property returning -1.